### ISSUE TITLE
Getting Zeros from target generator for cls_targets, ctr_targets and box_targets

### ISSUE THREAD

**Issue by wms2537** [Angzz/fcos-gluon-cv#2]
`[0. 0. 0. ... 0. 0. 0.]
<NDArray 20938 @cpu(0)>	
[0. 0. 0. ... 0. 0. 0.]
<NDArray 20938 @cpu(0)>	
[[0. 0. 0. 0.]
 [0. 0. 0. 0.]
 [0. 0. 0. 0.]
 ...
 [0. 0. 0. 0.]
 [0. 0. 0. 0.]
 [0. 0. 0. 0.]]
<NDArray 20938x4 @cpu(0)>	
[[  0.   0.]
 [128.   0.]
 [256.   0.]
 ...
 [976. 992.]
 [984. 992.]
 [992. 992.]]
<NDArray 20938x2 @cpu(0)>`
I'm getting this from the target generator for cls_targets, ctr_targets, box_targets, cor_targets. My input image is (1000, 1000, 3) and the boxes are [291, 114, 901, 778, 60],  [504, 263, 780, 490, 15], [461, 222, 829, 579, 20], [ 24, 205, 389, 800, 15]
What is the possible cause for this? 
I'm using resnet50 as the base.

**Comment by Angzz** [comment]
Hi, did you mean all zeros for the cls_targets, we follow your settings and run the following code, and the result seems normal.
 
```
img = nd.zeros(shape=(1000, 1000, 3))                    
boxes = nd.array([[291, 114, 901, 778, 60], [504, 263, 780, 490, 15],              
                             [461, 222, 829, 579, 20],  [24, 205, 389, 800, 15]], ctx=mx.cpu())                                                    
target_generator = FCOSTargetGenerator()                 
cls_targets, ctr_targets, box_targets, cor_targets = target_generator.generate_targets(img, boxes) 
print((cls_targets != 0).sum()) # 101
print((ctr_targets != 0).sum()) # 101
```

**Comment by wms2537** [comment]
But the model does not seem to converge after a few epochs. The mAp stays at  around 0.001

**Comment by Angzz** [comment]
we only train the model on coco dataset, and the results are slightly lower
than the official version, for other datasets, you can change the warm
iter, and dd you use the GN version? we find it make the training process
more stable.

wms2537 <notifications@github.com> 于2019年5月27日周一 下午7:46写道：

**Comment by wms2537** [comment]
May I know what is the initial loss of your model? Or could you post your training plot ?

**Comment by Angzz** [comment]
hi, here is my training loss.
[image: image.png]

wms2537 <notifications@github.com> 于2019年5月31日周五 下午6:13写道：
